Improve Clarity of the `Build forms with API` Recipe (#5899)
withastro/docs: src/content/docs/en/recipes/build-forms-api.mdx

## Recipe

> 1. First, create a `POST` API endpoint at `/api/feedback` that will receive the form data. Use `request.formData()` to process it. Be sure to validate the form values before you use them. 

Reviewer: ```suggestion
1. Create a `POST` API endpoint at `/api/feedback` that will receive the form data. Use `request.formData()` to process it. Be sure to validate the form values before you use them. 
```

    </UIFrameworkTabs>

> 3. Now, create a function that accepts a submit event, then pass it as a `submit` handler to your form. In the function, call `preventDefault` on the event to override the browser's default submission process. 

Reviewer: ```suggestion
3. Create a function that accepts a submit event, then pass it as a `submit` handler to your form. 

    In the function:
    - Call `preventDefault()` on the event to override the browser's default submission process.
    - Create a `FormData` object and send it in a `POST` request to your endpoint using `fetch()`.
```

>     Then, create a `FormData` object and send it in a `POST` request to your endpoint using `fetch`. 

Reviewer: ```suggestion
```

    <UIFrameworkTabs>
      <Fragment slot="preact">

>         ```tsx title="src/components/FeedbackForm.tsx" "/api/feedback" add={1, 4-17, 34} add="onSubmit={submit}"

Reviewer: ```suggestion
        ```tsx title="src/components/FeedbackForm.tsx" "/api/feedback" add={1, 4-17, 34} add="onSubmit={submit}" "formData" "e.preventDefault();"
```

Author: Shall I do this for all the frameworks?

Reviewer: If you think it looks good, then yeah!

        <script setup lang="ts">
        import { ref } from "vue";

        const responseMessage = ref<string>();

        async function submit(e: Event) {
          e.preventDefault();
          const formData = new FormData(e.currentTarget as HTMLFormElement);
          const response = await fetch("/api/feedback", {
            method: "POST",
            body: formData,
          });
          const data = await response.json();
          responseMessage.value = data.message;
        }
        </script>

        <template>
          <form @submit="submit">
            <label>
              Name
              <input type="text" id="name" name="name" required />
            </label>
            <label>
              Email
              <input type="email" id="email" name="email" required />
            </label>
            <label>
              Message
              <textarea id="message" name="message" required />
            </label>
            <button>Send</button>
            <p v-if="responseMessage">{{ responseMessage }}</p>
          </form>
        </template>
        ```
      </Fragment>

    </UIFrameworkTabs>

> 4. Test out your new form component by importing and displaying it on a page! Be sure to use a `client:*` directive to ensure that the form logic is hydrated when you want it to be.

Reviewer: ```suggestion
4. Import and include your `<FeedbackForm />` component on a page. Be sure to use a `client:*` directive to ensure that the form logic is hydrated when you want it to be.
```

          const response = await fetch("/api/feedback", {
            method: "POST",
            body: formData,
          });
          const data = await response.json();
          responseMessage.value = data.message;
        }
        </script>

        <template>
          <form @submit="submit">
            <label>
              Name
              <input type="text" id="name" name="name" required />
            </label>
            <label>
              Email
              <input type="email" id="email" name="email" required />
            </label>
            <label>
              Message
              <textarea id="message" name="message" required />
            </label>
            <button>Send</button>
            <p v-if="responseMessage">{{ responseMessage }}</p>
          </form>
        </template>
        ```
      </Fragment>

    </UIFrameworkTabs>

4. Import and include your `<FeedbackForm />` component on a page. Be sure to use a `client:*` directive to ensure that the form logic is hydrated when you want it to be.
    
    <UIFrameworkTabs>
        <Fragment slot="preact">
        ```astro title="src/pages/index.astro" "client:load"

>         ---
>         import FeedbackForm from "../components/FeedbackForm"
>         ---

Author: ```suggestion
        ---
        export const prerender = false;
        import FeedbackForm from "../components/FeedbackForm"
        ---
```

Author: @sarah11918 if preferred, I can add this to the code snippet at the end as well.

Reviewer: Ah, I guess the *page itself* doesn't need to be on-demand rendered as long as the API endpoint is. So, in fact, in hybrid mode, shouldn't `feedback.ts` need `export const prerender = false;`?

Reviewer: SO, if having the statement doesn't actually make a difference, let's not put it in and... see what happens!  :grinning:
